frontend/src/component/pages/shared/UserShortLinksSection.spec.tsx

```diff
@@ -0,0 +1,190 @@
+    expect(container.textContent).not.toContain('Long URL');
+    expect(container.textContent).not.toContain('Alias');
+  });
+
+  test('should render when "total" attribute in pagedShortLinks is non zero with empty shortLinks', () => {
+    const pagedShortLinks: IPagedShortLinks = {
+      shortLinks: [],
+      totalCount: 1
+    };
+
+    const { container } = render(
+      <UserShortLinksSection
+        onPageLoad={jest.fn}
+        pagedShortLinks={pagedShortLinks}
+      />
+    );
+
+    expect(container.textContent).toContain('Created Short Links');
+    expect(container.textContent).toContain('Long URL');
+    expect(container.textContent).toContain('Alias');
+  });
+
+  test('should render passed short links correctly', () => {
+    const pagedShortLinks: IPagedShortLinks = {
+      shortLinks: sampleShortLinks.slice(0, 3),
+      totalCount: sampleShortLinks.length
+    };
+
+    const { container } = render(
+      <UserShortLinksSection
+        onPageLoad={jest.fn}
+        pagedShortLinks={pagedShortLinks}
+      />
+    );
+
+    expect(container.textContent).toContain('Created Short Links');
+    expect(container.textContent).toContain('Long URL');
+    expect(container.textContent).toContain('Alias');
+
+    for (let urlIdx = 0; urlIdx < pagedShortLinks.shortLinks.length; urlIdx++) {
+      const shortLink = pagedShortLinks.shortLinks[urlIdx];
+      expect(container.textContent).toContain(shortLink.originalUrl);
+      expect(container.textContent).toContain(shortLink.alias);
+    }
+  });
+
+  test('should render correctly when total short links passed is less than page size', () => {
+    const pagedShortLinks: IPagedShortLinks = {
+      shortLinks: sampleShortLinks.slice(0, 3),
+      totalCount: 3
+    };
+    const pageSize = 5;
+
+    const { container } = render(
+      <UserShortLinksSection
+        onPageLoad={jest.fn}
+        pagedShortLinks={pagedShortLinks}
+        pageSize={pageSize}
+      />
+    );
+
+    expect(container.querySelectorAll('.page-number')).toHaveLength(1);
+  });
+
+  test('should render multiple pages when total short link count exceeds page size', () => {
+    const pageSize = 5;
+    const pagedShortLinks: IPagedShortLinks = {
+      shortLinks: sampleShortLinks.slice(0, pageSize),
+      totalCount: sampleShortLinks.length
+    };
+
+    const { container } = render(
+      <UserShortLinksSection
+        onPageLoad={jest.fn}
+        pagedShortLinks={pagedShortLinks}
+        pageSize={pageSize}
+      />
+    );
+
+    const expectedPagesCount = Math.ceil(pagedShortLinks.totalCount / pageSize);
+    expect(container.querySelectorAll('.page-number')).toHaveLength(
+      expectedPagesCount
+    );
+  });
+
+  test('should request short links for initial page correctly', () => {
+    const onPageLoad = jest.fn();
+
+    render(<UserShortLinksSection onPageLoad={onPageLoad} />);
+
+    expect(onPageLoad).toBeCalledTimes(1);
```

Comment:
We should provide fake data to this function and check rendering

Reply:
@byliuyang , I doubt if we should test for the what the handle function is doing to our component. We have already covered the case if the component renders the passed props as expected. This test would cover while navigating, if appropriate function with expected params are being called. `onPageLoad` re-renders the component with the updated prop.

https://stackoverflow.com/questions/54479814/how-to-test-props-that-are-updated-by-an-onchange-handler-in-react-testing-libra

Reply:
https://martinfowler.com/articles/mocksArentStubs.html#TheDifferenceBetweenMocksAndStubs

Here is the preference of test doubles for unit testing:
Real implementation > Fake > Stub > Mock
We use fake when the real-implementation is slow or is hardware devices. We use stub when fake is overly complex to do. We use mock when interacting with external systems that we don't have control of.
https://blog.pragmatists.com/test-doubles-fakes-mocks-and-stubs-1a7491dfa3da